hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+          name:  "BYTE_A",
+          desc:  '''One of four bytes to transmit.  If ByteOrder = 0,
+                    This byte will be transmitted last, after BYTE_D, BYTE_C
+                    and BYTE_B.  Otherwise BYTE_A will be transmitted first.
+                    ''',
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "RXDATA",
+      desc: "SPI Receive data",
+      swaccess: "ro",
+      hwaccess: "hwo",
+      fields: [
+        { bits:  "31:24",
+          name:  "BYTE_D",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    this byte will be received before BYTE_C, BYTE_B
+                    and BYTE_A. Otherwise BYTE_D will be received
+                    last of the four.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "23:16",
+          name:  "BYTE_C",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    this byte will be received second, after BYTE_D.
+                    Otherwise BYTE_C will be received third, after
+                    BYTE_A, and BYTE_B.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "15:8",
+          name:  "BYTE_B",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    this byte will be received third after BYTE_D
+                    and BYTE_C.  Otherwise BYTE_B will be received
+                    second, after BYTE_A.
+                    ''',
+          resval: "0x0"
+        },
+        { bits:  "7:0",
+          name:  "BYTE_A",
+          desc:  '''One of four bytes to receive.  If ByteOrder = 0,
+                    This byte will be received last, after BYTE_D, BYTE_C
+                    and BYTE_B.  Otherwise BYTE_A will be received first.
+                    ''',
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "ERROR_ENABLE",
+      desc: "Controls which classes of errors raise an interrupt."
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          name: "CMDERR",
+          desc: '''Command Error: Assert to send an error interrupt whenever
+                   a command is issued while BUSY is asserted'''
+          resval: "0x1"
+        },
+        { bits: "1",
+          name: "OVERFLOW",
+          desc: '''Overflow Errors: Assert to send an error interrupt whenever
+                   the TXFIFO overflows.'''
+          resval: "0x1"
+        },
+        { bits: "2",
+          name: "UNDERFLOW",
+          desc: '''Underflow Errors: Assert to send an error interrupt whenever
+                   the RXFIFO underflows.'''
+                   resval: "0x1"
+        },
+      ]
+    },
+    { name: "ERROR_STATUS",
+      desc: '''Indicates that any errors that have occurred.  When an error
+               occurs, the corresponding bit must be cleared here before
+               issuing any further commands.'''
```

Comment:
This looks reasonably awful in the generated documentation, but I think we should fix that in the documentation generator, not here. Maybe add a blank line and we'll treat the first line as a heading and follow-on text as markdown paragraphs? (I'll file a reggen PR to do this later)

Reply:
Multi-paragraph descriptions should be supported once #5116 is merged.

Reply:
I don't think I understand what you are asking me to do with this.  Could you please explain.

Reply:
Ok... after locally rebasing #5116 on top of this PR, I see what you mean. That is an improvement.  Thanks!
I'll add a commit for the changes.